pkg/logging/config.go

```diff
@@ -37,10 +37,10 @@ const (
 // If configuration is empty, a fallback configuration is used.
 // If configuration cannot be used to instantiate a logger,
 // the same fallback configuration is used.
-func NewLogger(configJSON string, levelOverride string) (*zap.SugaredLogger, zap.AtomicLevel) {
+func NewLogger(configJSON string, levelOverride string) (*zap.SugaredLogger, zap.AtomicLevel, error) {
```

Comment:
This signature seems a little weird. Even when we return an error, there is still a logger.

May be we can simplify this whole thing: Remove all info level logs from this file and keep the error log as is, since it is something we probably should see anyway. Or alternatively, remove all logging from this file and simply tag the logger that is returned with something like "default-logger" if we failed to create the logger with the user supplied parameters.

Third option would be to pass in a simple bool to this method to say whether it should log or not and if true, it logs all info and error, and if false, doesnt log anything. Unit tests can set bool to false and prod can set it to true. My preference would be this third option without making any other changes.

Reply:
@grantr proposed a fourth option - rename this method to lets say foo. Create a package level function that is set to foo by default. Unit tests can create a different "NewLogger" method and set the package level variable to this. So, nothing except unit tests change.

Reply:
I like the 4th option because it makes the code more readable and also from logs an operator can understand that default logger is initiated.